app/controllers/books_controller.rb

```diff
@@ -56,4 +61,8 @@ def set_book
   def book_params
     params.require(:book).permit(:name, :description, :release)
   end
+
+  def update_all_key
+    Rails.cache.write('all_key_books', Book.all.to_a)
```

Comment:
Option: just remove key cache from redis, so that data will be called from DB once and cache again with correctness